### PR TITLE
Close #21 create func compute centiloid

### LINKED ISSUE #21: Feature Request: Create compute_centiloid()
### Feature Idea

Make a function called compute_centiloid() to generate centiloid values given amyloid PET tracer (ag_neuro), SUVr pipeline and reference (nv_neuro and suppnv_neuro).

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by jwang-lilly**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/compute_centiloid.R [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+#' Centiloid is calculated as:
+#'
+#' Centiloid = m * SUVR + c
+#'
+#' where m and c are formula parameters. If `custom_params` is provided, this formula
+#' is used regardless of the tracer, pipeline, or reference region values.
+#'
+#' If a matching standard combination is not found and no `custom_params` are provided,
+#' a warning is issued and `NA_real_` is returned.
+#'
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:53]: Can you please in some way include references and formulas for each of the possilble combinations of `m` and `c` that are pre-specified in `tracer`, `pipeline`, `ref_region`?

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Hi @jwang-lilly,

Thanks for this! It's a great start. Please find some initial comments below regarding how users should interface with the function - then can review tests and function code later:

- I am wondering if we could simplify the arguments and input testing. Specifically, I wonder if we really _need_ `m` and `c` to be specified within a `custom_params` list? It doesn't really match the other admiral `compute_*|` functions. What about something like:
```
compute_centiloid <- function(
  tracer, 
  pipeline, 
  ref_region,
  suvr, 
  custom_m,
  custom_c
)
````
with the input validation logic and order as follows:
  - Make all arguments optional (`optional = TRUE` in the `assert_*` functions)
  - Pre-specified method: check if `tracer`, `pipeline`, `ref_region` are all not `NULL`, if so check that `custom_m` and `custom_c` are null and `tracer`, `pipeline`, `ref_region`  are from accepted set of values. Then set `m` and `c` to the prespecified values based on the other three parameters.
  - Custom formula: check if `custom_m`, `custom_c` are both not `NULL`, if so check that `tracer`, `pipeline`, `ref_region` are all `NULL`.  Then set `m` and `c` to `custom_m` and `custom_c`.
  - Proceed with function calculation from there.

Additionally, please could you delete the dummy function file in the R folder now that we have a real function on the way? Thanks!

File: R/compute_centiloid.R (hunk 1 above)
Comment (jwang-lilly) [reply R/compute_centiloid.R:59]: Thanks!

File: R/compute_centiloid.R [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+#'   If custom slope and intercept are provided, any string is accepted.
+#'
+#' @param suvr SUVR value
+#'
+#'   A numeric value is expected.
+#'
+#' @param custom_slope Optional slope parameter for custom centiloid calculation formula
+#'
+#'   A numeric value is expected when provided.
+#'   When provided with custom_slope, this overrides the standard formula parameters.
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:35]:
```suggestion
#'   When `custom_slope` is specified (along with `custom_intercept`), this overrides the standard formula parameters `tracer, `pipeline` and `ref_region`.
```

File: R/compute_centiloid.R [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+#'   Default is NULL.
+#'
+#' @param custom_intercept Optional intercept parameter for custom centiloid calculation formula
+#'
+#'   A numeric value is expected when provided.
+#'   When provided with custom_intercept, this overrides the standard formula parameters.
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:41]:
```suggestion
#'   When `custom_intercept` is specified (along with `custom_slope`), this overrides the standard formula parameters `tracer, `pipeline` and `ref_region`.
```

File: R/compute_centiloid.R [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+#'   Default is NULL.
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:42]:
```suggestion
#'   Default is `NULL`.
```

File: R/compute_centiloid.R [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+#'
+#' @param suvr SUVR value
+#'
+#'   A numeric value is expected.
+#'
+#' @param custom_slope Optional slope parameter for custom centiloid calculation formula
+#'
+#'   A numeric value is expected when provided.
+#'   When provided with custom_slope, this overrides the standard formula parameters.
+#'   Default is NULL.
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:36]:
```suggestion
#'   Default is `NULL`.
```

File: R/compute_centiloid.R [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+#'
+#' @details
+#' The centiloid scale is a standardized quantitative measure for amyloid PET imaging
+#' that allows comparison between different tracers and analysis methods. This function
+#' converts SUVR values to the centiloid scale based on published conversion
+#' equations for specific tracer, pipeline, and reference region combinations.
+#'
+#' Centiloid is calculated as:
+#'
+#' Centiloid = slope * SUVR + intercept
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:52]: This would look nicer if formatted like an equation (see [here](https://github.com/pharmaverse/admiral/blob/976de6a852ca94e68f190ea64cd36d4b998db435/R/compute_kidney.R#L52) for an example)

```suggestion
#' \deqn{Centiloid = slope \times SUVR + intercept}
```

File: R/compute_centiloid.R [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+#'
+#' where slope and intercept are formula parameters. If custom slope and intercept are provided,
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:54]:
```suggestion
#' where slope and intercept are formula parameters. If `custom slope` and `custom_intercept`
#' are not provided, this function uses pre-defined values for these constants based on the user's 
#' selections of tracer, pipeline and reference region. 
#'
#' <Add details (+ references) for all accepted combinations of tracer, pipeline and ref region>
#';
#' Alternatively, the user can override the 
#' pre-selection by specifying `custom_slope` and `custom_intercept` instead.
```

File: R/compute_centiloid.R [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+#' this formula is used regardless of the tracer, pipeline, or reference region values.
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:55]:
```suggestion
```

File: R/compute_centiloid.R [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+#' )
+compute_centiloid <- function(
+    tracer,
+    pipeline,
+    ref_region,
+    suvr,
+    custom_slope = NULL,
+    custom_intercept = NULL) {
+  params_map <- list(
+    # reference: https://doi.org/10.1016/j.jalz.2018.06.1353
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:93]: Ok to have references here, but these comments are not exposed to the users unless they peruse the actual function code (unlikely). As such, please add each of the three references to the `@details` section as well, as the roxygen documentation is exposed to the users.

File: R/compute_centiloid.R [hunk 10]
```diff
@@ -0,0 +1,176 @@
+#' Compute Centiloid Value
+#'
+#' Computes centiloid values based on amyloid PET tracer, SUVR value and pipeline,
+#' and reference region.
+#' Also allows for custom formula parameters.
+#'
+#' @param tracer Amyloid PET tracer from ag_neuro dataset
+#'
+#'   A character string is expected.
+#'   When custom parameters are not provided, accepted values are:
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:10]: As I understand it, not all accepted values are actually valid, depending what the values of `pipeline` and `ref_region` are. So, I would suggest making a list of accepted combinations in the details section instead, and have left the same comment for the other two parameters.

```suggestion
#'  See Details section for accepted values in combination with `pipeline` and `ref_region`.
```

File: R/compute_centiloid.R [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+#'   '18F-Florbetapir', '18F-Florbetaben'
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:11]:
```suggestion
```

File: R/compute_centiloid.R [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+#'   If custom slope and intercept are provided, any string is accepted.
+#'
+#' @param pipeline SUVR pipeline from nv_neuro dataset
+#'
+#'   A character string is expected.
+#'   When custom parameters are not provided, accepted values are:
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:17]:
```suggestion
#'  See Details section for accepted values in combination with `tracer` and `ref_region`.
```

File: R/compute_centiloid.R [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+#'   'AVID FBP SUVR PIPELINE', 'BERKELEY FBP SUVR PIPELINE', 'BERKELEY FBB SUVR PIPELINE'
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:18]:
```suggestion
```

File: R/compute_centiloid.R [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+#'   If custom slope and intercept are provided, any string is accepted.
+#'
+#' @param ref_region Reference region from suppnv_neuro dataset
+#'
+#'   A character string is expected.
+#'   When custom parameters are not provided, accepted values are:
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:24]:
```suggestion
```suggestion
#'  See Details section for accepted values in combination with `tracer` and `pipeline`.
```
```

File: R/compute_centiloid.R [hunk 15]
```diff
@@ -0,0 +1,176 @@
...
+#'   'Whole Cerebellum', 'Composite Reference Region'
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:25]:
```suggestion
```

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Hi @jwang-lilly thanks for the updates - I have now carefully reviewed the roxygen documentation and provided some comments. I also have some comments about the actual code, but didn't want to leave too many comments at once, so will add those later once the documentation conversations are resolved. Hope that's ok!

**Comment by manciniedoardo** [comment]
@jwang-lilly Actually, I condensed my comments about the function in one general comment - see below. Will add now as it's fresh in my mind, but happy to discuss more if need be.

___________________________________________________

I think the input validation and function logic can be greatly simplified. What about the following pseudocode?

- Check if either of `custom_slope` and `custom_intercept` are not missing and of right type. If so, input validation is only required for `custom_slope`, `custom_interecept` and `suvr`. No input validation needs to be done on `tracer`, `pipeline` and `ref_region` as these are ignored anyway. Within same "if" condition, set `slope` and `intercept` to values in `custom_slope` and `custom_intercept` respectively.

- Otherwise, check that `tracer`, `pipeline`, `ref_region` and `suvr` are all non missing and of right type. Within same "else" condition, compare across an existing dataframe of valid combinations (you could even add a copy of this table in the details section above). Something like:

```
valid_combinations <- tribble(
  ~tracer,  ~pipeline, ~ref_region, ~slope, ~intercept,
  "Type 1", "Type 3",  "Type 2",    5,      1,
  "Type 1", "Type 1",  "Type 3",    4,      2,
  "Type 2", "Type 3",  "Type 1",    3,      3
)

check <- valid_combinations %>% 
  filter(
    tracer == !!tracer,
    pipeline == !!pipeline,
    ref_region == !!ref_region
  )

if(nrow(check) == 0){
  <throw error>
}else{
  slope <- check$slope
  intercept <- check$intercept
}
```

- calculate centiloid

Note: when throwing errors/warnings, please use the {admiral}-standard {cli} package rather than `stop()`/`warning()`. I am happy to help with this if confusing - just lmk!

**Comment by jwang-lilly** [comment]
Many thanks to @manciniedoardo for helpful comments and suggestions.

File: R/compute_centiloid.R [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+        tracer == !!tracer,
+        pipeline == !!pipeline,
+        ref_region == !!ref_region
+      )
+
+    if (nrow(check) == 0) {
+      cli::cli_warn(
+        c(
+          "No standard conversion formula available for:",
+          "i" = "tracer = {.val {tracer}}",
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:166]: Errors look great now! Minor note: I wonder if these would look better as bullets (*)rather than i's? Leaving up to you though, feel free to take or leave

File: tests/testthat/test-compute_centiloid.R [hunk 17]
```diff
@@ -0,0 +1,204 @@
+# ---- Valid combination tests ----
```
Comment (manciniedoardo) [inline tests/testthat/test-compute_centiloid.R:1]: Great job with these tests! We can even explore snapshot testing for the error/warnings, but it might be a bit overkill. Let's discuss next week and not let it stop merging this PR.

File: R/compute_centiloid.R [hunk 18]
```diff
@@ -0,0 +1,181 @@
...
+compute_centiloid <- function(
+    tracer,
+    pipeline,
+    ref_region,
+    suvr,
+    custom_slope = NULL,
+    custom_intercept = NULL) {
+  valid_combinations <- tibble::tribble(
+    ~tracer,              ~pipeline,                     ~ref_region,          ~slope,   ~intercept,
+    "18F-Florbetapir",    "AVID FBP SUVR PIPELINE",      "Whole Cerebellum",   183,      -177,
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:118]: minor note: I wonder if we want to be consistent with the casing? as pipeline is in all caps but others are not? leaving up to you.

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
@jwang-lilly Looking awesome! I left a few final comments but it's 90% there, you did a super job 😄

File: R/compute_centiloid.R [hunk 19]
```diff
@@ -0,0 +1,181 @@
...
+    "18F-Florbetapir",    "BERKELEY FBP SUVR PIPELINE",  "Whole Cerebellum",   188.22,   -189.16,
+    "18F-Florbetaben",    "BERKELEY FBB SUVR PIPELINE",  "Whole Cerebellum",   157.15,   -151.87
+  )
+
+  # Check custom_slope and custom_intercept
+  has_custom_slope <- !is.null(custom_slope)
+  has_custom_intercept <- !is.null(custom_intercept)
+
+  if (has_custom_slope != has_custom_intercept) {
+    cli::cli_abort("Both {.code custom_slope} and {.code custom_intercept}
```
Comment (manciniedoardo) [inline R/compute_centiloid.R:128]: Across the actual function text, our convention in admiral is not to use `::` but instead add the functions to the imported list in `admiralneuro-package.R`

File: R/compute_centiloid.R [hunk 20]
```diff
@@ -0,0 +1,181 @@
...
+#' Centiloid is calculated as:
+#'
+#' Centiloid = \deqn{slope \times SUVR + intercept}
+#'
+#' where slope and intercept are formula parameters. If `custom_slope` and `custom_intercept`
+#' are not specified, this function uses pre-defined slope and intercept based on
+#' the user's selections of tracer, pipeline, and reference region.
+#
+#' The following combinations of tracer, pipeline, and reference region are supported:
+#'
```
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply R/compute_centiloid.R:58]: https://stackoverflow.com/questions/42813243/insert-markdown-table-in-roxygen2-r-package-documentation

**Review by manciniedoardo — APPROVED** [review]
@jwang-lilly this is awesome! great job!

**Comment by manciniedoardo** [comment]
@pharmaverse/admiralneuro please take a look and review from a neuro standpoint too!

**Review by meilinjiang — APPROVED** [review]
I have carefully reviewed form the neuro perspective and it looks great. My only comment is that it would be more user-friendly if we can shorten the input length of accepted values for "pipeline". For example, we could denote "AVID FBP SUVR PIPELINE" as something like "FBP_AVID" and spell out the full name "AVID FBP SUVR PIPELINE" in the Details section. Similarly for other Berkeley pipelines.

[Somehow I couldn't get the automatic notification about updates to my work email. It doesn't allow me to add my work email. I will try to figure it out.]

**Comment by manciniedoardo** [comment]
My only comment is that it would be more user-friendly if we can shorten the input length of accepted values for "pipeline". For example, we could denote "AVID FBP SUVR PIPELINE" as something like "FBP_AVID" and spell out the full name "AVID FBP SUVR PIPELINE" in the Details section. Similarly for other Berkeley pipelines.

That's a great point, @jwang-lilly can you take a look please?
